clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -304,6 +306,12 @@ public synchronized void bootstrap(List<InetSocketAddress> addresses) {
         this.needFullUpdate = true;
         this.updateVersion += 1;
         this.metadataSnapshot = MetadataSnapshot.bootstrap(addresses);
+        this.bootstrapAddresses = addresses;
```

Comment:
Is there a reason we are re-bootstrapping from addresses that were resolved when the client was created? This seems different from the example in the KIP where the addresses that the bootstrap resolved to had changed.

Reply:
This really depends on the `client.dns.lookup`. E.g. if you use the bootstrap address `kafka.myorg.com:12345`, the bootstrap node will get `kafka.myorg.com` as the host, despite the name resolution happens before that.

Reply:
Yes, this seems fine because we are only using host string from the InetSocketAddress to build the bootstrap cluster.